code/src/sixsq/nuvla/server/util/kafka.clj

```diff
@@ -1,12 +1,42 @@
 (ns sixsq.nuvla.server.util.kafka
+  "Kafka client can be configured using KAFKA_CLIENT_CONF_xxx env vars. For
+  example to set 'delivery.timeout.ms' to 12345, provide it as the following
+  environment variable: KAFKA_CLIENT_CONF_DELIVERY_TIMEOUT_MS=12345"
   (:require
     [clojure.tools.logging :as log]
     [environ.core :as env]
-    [kinsky.client :as kc]))
+    [kinsky.client :as kc]
+    [sixsq.nuvla.utils.env :as eu]
+    [clojure.string :as str]))
 
 
 (def ^:dynamic *producer* nil)
 
+(def ^:const warn-producer-no-initialised "Kafka producer is not initialized.")
+
+(def ^:const kafka-client-conf-prefix ":kafka-client-conf-")
+
+(def ^:const producer-name-default "nuvla-server")
+
+
+(defn client-params-from-env
+  "Given a map of env vars 'env-vars', return a map with Kafka only params
+  filtered out by 'prefix', and keys transformed into Kafka client params.
+  Example of the transformation:
+  Environment var: KAFKA_CLIENT_CONF_DELIVERY_TIMEOUT_MS=12345 ->
+  Clojure env var: {:kafka-client-conf-delivery-timeout-ms 12345} ->
+  Kafka parameter: {:delivery.timeout.ms 12345}"
+  [env-vars prefix]
+  (->> env-vars
+       (filter #(str/starts-with? (-> % first str) prefix))
+       (map #(identity [(-> %
```

Comment:
Small change, we don't need identity we can build a vector with following function
```
       (map (fn [[k v]]
              (vector (-> (str k)
                          (str/replace (re-pattern prefix) "")
                          (str/replace #"-" ".")
                          keyword) v)))
```